File: LowLevelDesign/tic-tac-toe/src/Cell.ts
import { Symbol } from "./enum/Symbol";
export class Cell {
    private symbol: Symbol;  
    constructor(public row: number, public col: number) {
        this.row = row;
        this.col = col;
        this.symbol = Symbol.EMPTY;
    }

    public getSymbol(): Symbol {
        return this.symbol;
    }

    public updateCell(symbol: Symbol): void {
        if (this.symbol === Symbol.EMPTY) {
            this.symbol = symbol;
        } else {
            throw new Error("Cell is already occupied");
        }
    }
}

File: SystemDesignDiagrams/practice/LLD/tictactoe/src/main/java/com/tictactoe/entities/Cell.java
package com.tictactoe.entities;

import com.tictactoe.enums.Symbol;

public class Cell {
    private final int row;
    private final int col;
    private Symbol symbol;

    public Cell(int row, int col) {
        this.row = row;
        this.col = col;
    }

    public void updateCell(Symbol symbol) {
        this.symbol = symbol;
    }

    public Symbol getSymbol() {
        return this.symbol;
    }
}
